File: src/main/java/org/veupathdb/lib/blast/field/NumAlignments.java
package org.veupathdb.lib.blast.field;

import com.fasterxml.jackson.databind.JsonNode;
import org.veupathdb.lib.blast.util.DefaultingJSONValue;
import org.veupathdb.lib.blast.util.JSONConstructor;

public record NumAlignments(long value) implements DefaultingJSONValue
{
  public static final long DefaultValue = 250;

  @Override
  public boolean isDefault() {
    return value == DefaultValue;
  }

  @Override
  public JsonNode toJSON() {
    return JSONConstructor.newLong(value);
  }
}

File: src/main/kotlin/org/veupathdb/lib/blast/field/num-alignments.kt
package org.veupathdb.lib.blast.field

import com.fasterxml.jackson.databind.node.ObjectNode
import org.veupathdb.lib.blast.serial.BlastField


private const val KeyNumAlignments     = "-num_alignments"
private const val DefaultNumAlignments = 250L


internal fun ParseNumAlignments(js: ObjectNode) =
  NumAlignments(js[KeyNumAlignments]?.longValue() ?: DefaultNumAlignments)


@JvmInline
value class NumAlignments(val value: Long = DefaultNumAlignments) : BlastField {
  override val isDefault get() = value == DefaultNumAlignments

  override fun appendJson(json: ObjectNode) {
    if (!isDefault)
      json.put(KeyNumAlignments, value)
  }

  override fun appendCliSegment(cli: StringBuilder) {
    if (!isDefault)
      cli.append(' ').append(KeyNumAlignments).append(' ').append(value)
  }

  override fun appendCliParts(cli: MutableList<String>) {
    if (!isDefault) {
      cli.add(KeyNumAlignments)
      cli.add(value.toString())
    }
  }
}